lib/shopify-cli/command.rb

```diff
@@ -41,13 +43,16 @@ def subcommand_registry
       end
 
       def prerequisite_task(*tasks)
-        tasks.each do |task|
-          prerequisite_tasks[task] = ShopifyCli::Tasks::Registry[task]
-        end
+        @prerequisite_tasks ||= []
+        @prerequisite_tasks += tasks
+      end
+
+      def run_prerequisites
+        (@prerequisite_tasks || []).each { |task| task_registry[task]&.call(@ctx) }
       end
 
-      def prerequisite_tasks
-        @prerequisite_tasks ||= {}
+      def task_registry
+        @task_registry || ShopifyCli::Tasks::Registry
```

Comment:
Should this be a `||=` instead of just a `||`?

Reply:
This is not an assignment, @task_registry can already be set from `attr_writer` so this really only allows a default if `@task_registry` is nil